### PR TITLE
fix a bug that nullable unique index can not find null values using index.

### PR DESCRIPTION
**Pull request by winkyao**
## What have you changed? (mandatory)

Append 0 to the EndKey of the 'Is Null' index range. To make unique-key range [null, null] to be a scan rather than point-get in tikv.

See https://github.com/pingcap/tidb/blob/master/store/mockstore/mocktikv/executor.go#L289, and it will judge the range as point-get by:

```
// IsPoint checks if the key range represents a point.
func (r *KeyRange) IsPoint() bool {
	return bytes.Equal(r.StartKey.PrefixNext(), r.EndKey)
}
```

## What is the type of the changes? (mandatory)

- Bug fix (non-breaking change which fixes an issue)

## How has this PR been tested? (mandatory)

UT

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

no

## Does this PR affect tidb-ansible update? (mandatory)

no

## Does this PR need to be added to the release notes? (mandatory)

no

### CODE REVIEW

**Comment by winkyao** [comment]
@coocood @winoros PTAL

File: distsql/request_builder.go [hunk 1]
```diff
@@ -314,8 +314,21 @@ func encodeIndexKey(sc *stmtctx.StatementContext, ran *ranger.Range) ([]byte, []
...
 	}
+
 	if !ran.HighExclude {
 		high = []byte(kv.Key(high).PrefixNext())
 	}
+
+	var hasNull bool
+	for _, highVal := range ran.HighVal {
+		if highVal.IsNull() {
+			hasNull = true
```
Comment (winoros) [inline distsql/request_builder.go]: We could break at once?

File: util/ranger/types.go [hunk 2]
```diff
@@ -67,6 +67,10 @@ func (ran *Range) IsPoint(sc *stmtctx.StatementContext) bool {
 		if cmp != 0 {
 			return false
 		}
+
+		if a.IsNull() && b.IsNull() {
```
Comment (winoros) [inline util/ranger/types.go]: Since `cmp` is 0. Compare once is enough?

File: distsql/request_builder.go [hunk 3]
```diff
@@ -314,8 +314,21 @@ func encodeIndexKey(sc *stmtctx.StatementContext, ran *ranger.Range) ([]byte, []
...
+		}
+	}
+
+	if hasNull {
+		// append 0 to make unique-key range [null, null] to be a scan rather than point-get.
```
Comment (winoros) [inline distsql/request_builder.go]: Capitalize first character.

**Comment by winkyao** [comment]
@coocood PTAL

**Comment by coocood** [comment]
LGTM

**Pull request merged**